### PR TITLE
[REVIEW] Fix string binop to update nulls appropriately 

### LINKED ISSUE #4264: [BUG] String binop doesn't update nulls
**Describe the bug**
When performing a string binop such as `==` cudf leaves the null rows untouched while pandas here returns `True/False` based on the condition.

**Steps/Code to reproduce bug**
```
import cudf
ser = cudf.Series(['a',None,'a',None])
ser == 'a'

0    True
1    null
2    True
3    null
dtype: bool
```

**Expected behavior**
```
import pandas as pd
ser = pd.Series(['a',None,'a',None])
ser == 'a'

0     True
1    False
2     True
3    False
dtype: bool
```

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: Dev-nightly
- Cudf @ 4460093650932fbf476dfd16194c51b07755325e

**Additional context**
This might probably go away once string ops are ported to the new api.

Simple workaround is to fill nulls with `True/False` based on the condition.

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
The basic code changes are in `binaryop.pyx` and `test_binops.py `, other changes are to go along with the requested behavior. `_align_to_index` wasn't actually aligning the indices in couple of scenarios which has been handled. This kind of addresses issue in #4709 , but we can get a better solution. Similarly `run` in  `applyutils.py` was creating columns as per input dataframe but wasn't creating the series with the same index, which is also taken care in this PR.

closes #4264

### CODE REVIEW

File: python/cudf/cudf/_libxx/binaryop.pyx [hunk 1]
```diff
@@ -150,6 +153,21 @@ cdef binaryop_s_v(Scalar lhs, Column rhs,
     return Column.from_unique_ptr(move(c_result))
 
 
+def handle_null_for_string_column(Column input_col, op):
+    if op in (
+        BinaryOperation['EQ'], BinaryOperation['LT'],
```
Comment (shwina) [inline python/cudf/cudf/_libxx/binaryop.pyx]: I think `op` here should just be a string (`eq`, `lt`, etc.,). The conversion to a `BinaryOperation` is extraneous.

Something that may help here: if `x` is a `BinaryOperation['SUM']` for example, then `x.name.lower() == 'sum'`.

File: python/cudf/cudf/core/series.py [hunk 2]
```diff
@@ -2505,7 +2505,16 @@ def _align_to_index(
                 raise ValueError("Cannot align indices with non-unique values")
         lhs = self.to_frame(0)
         rhs = cudf.DataFrame(index=as_index(index))
-        result = lhs.join(rhs, how=how, sort=sort)[0]
+        if how == "left":
+            lhs["____"] = cupy.arange(len(lhs))
```
Comment (shwina) [inline python/cudf/cudf/core/series.py]: Discussed offline and we're going to use `uuid4` here to ensure a unique column name

File: python/cudf/cudf/_libxx/binaryop.pyx [hunk 3]
```diff
@@ -154,14 +154,10 @@ cdef binaryop_s_v(Scalar lhs, Column rhs,
 
 
 def handle_null_for_string_column(Column input_col, op):
-    if op in (
-        BinaryOperation['EQ'], BinaryOperation['LT'],
-        BinaryOperation['LE'], BinaryOperation['GT'],
-        BinaryOperation['GE']
-    ):
+    if op.name.lower() in ('eq', 'lt', 'le', 'gt', 'ge'):
```
Comment (shwina) [inline python/cudf/cudf/_libxx/binaryop.pyx]: Just a minor nit here: let's make the input to the function a str, just for consistency with `binop` below.

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**